src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledAndTransitivePackageFeed.cs

```diff
@@ -25,59 +25,51 @@ public InstalledAndTransitivePackageFeed(IEnumerable<PackageCollectionItem> inst
         {
-            IEnumerable<ITransitivePackageReferenceContextInfo> transitivePRs = identity.PackageReferences.OfType<ITransitivePackageReferenceContextInfo>();
-
-            ITransitivePackageReferenceContextInfo transitivePR = transitivePRs.OrderByDescending(x => x.Identity.Version).FirstOrDefault();
+            cancellationToken.ThrowIfCancellationRequested();
 
-            if (transitivePR != null)
+            if (pkgIdentity is PackageCollectionItem identity)
             {
-                IReadOnlyCollection<PackageIdentity> transitiveOrigins = transitivePR.TransitiveOrigins?.Select(to => to.Identity).ToArray() ?? Array.Empty<PackageIdentity>();
+                IEnumerable<ITransitivePackageReferenceContextInfo> transitivePRs = identity.PackageReferences.OfType<ITransitivePackageReferenceContextInfo>();
 
-                if (transitiveOrigins.Any())
+                ITransitivePackageReferenceContextInfo transitivePR = transitivePRs.OrderByDescending(x => x.Identity.Version).FirstOrDefault();
+
+                if (transitivePR != null)
                 {
-                    // Get only local metadata. We don't want Deprecation and Vulnerabilities Metadata on Transitive packages
-                    IPackageSearchMetadata packageMetadata = await _metadataProvider.GetOnlyLocalPackageMetadataAsync(identity, cancellationToken);
+                    IReadOnlyCollection<PackageIdentity> transitiveOrigins = transitivePR.TransitiveOrigins?.Select(to => to.Identity).ToArray() ?? Array.Empty<PackageIdentity>();
 
-                    if (packageMetadata == null) // Edge case: local metadata not found
+                    if (transitiveOrigins.Any())
                     {
-                        packageMetadata = PackageSearchMetadataBuilder.FromIdentity(identity).Build(); // create metadata object only with ID
-                    }
+                        // Get only local metadata. We don't want Deprecation and Vulnerabilities Metadata on Transitive packages
+                        IPackageSearchMetadata packageMetadata = await _metadataProvider.GetOnlyLocalPackageMetadataAsync(identity, cancellationToken);
 
-                    var ts = new TransitivePackageSearchMetadata(packageMetadata, transitiveOrigins);
-                    return ts;
-                }
-                else
-                {
-                    return await base.GetPackageMetadataAsync(identity, includePrerelease, cancellationToken);
+                        if (packageMetadata == null) // Edge case: local metadata not found
+                        {
+                            packageMetadata = PackageSearchMetadataBuilder.FromIdentity(identity).Build(); // create metadata object only with ID
+                        }
+
+                        var ts = new TransitivePackageSearchMetadata(packageMetadata, transitiveOrigins);
+                        return ts;
```

Comment:
Variable assignment not needed. Never used, except for return.

```suggestion
                        return new TransitivePackageSearchMetadata(packageMetadata, transitiveOrigins);
```